bin/mocha

```diff
@@ -39,8 +39,9 @@ Object.keys(opts).forEach(opt => {
   }
 });
 
-// allow --debug to invoke --inspect on Node.js v8 or newer  nodeOpts.inspect = childOpts.debug;
+// allow --debug to invoke --inspect on Node.js v8 or newer
 if (childOpts.debug) {
+  nodeOpts.inspect = childOpts.debug;
   childOpts.timeout = false;
   delete childOpts.debug;
   debug('--debug -> --inspect');
```

Comment:
I don't understand where the distinction is made between Node >=v8 and Node v6.
We will deprecate `--debug` anyway once support of Node v6 is dropped, right? We could prepare this deprecation and pass the responsibility of choosing the correct Node flag to the user.

Reply:
Yeah, I agree, we should provide a deprecation message when debug is used outside of Node.js v6.

Reply:
actually I changed my mind.  a warning should be printed, but it's a nice convenience for users until Node.js v6 drops out of active maintenance.  at that point, we can remove it, because we'll drop v6 support as well.